### PR TITLE
Support arbitrary albumentations transforms

### LINKED ISSUE #995: Support arbitrary Albumentations augmentations
Currently, RV only supports a limited number of augmentations (listed below) and does not support setting their parameters.
```python3
augmentors = [
    'Blur', 'RandomRotate90', 'HorizontalFlip', 'VerticalFlip', 'GaussianBlur',
    'GaussNoise', 'RGBShift', 'ToGray'
]
```
It should be possible to support arbitrary augmentations without any serialization problems if we make use of Albumentation's [serialization utilities](https://github.com/albumentations-team/albumentations_examples/blob/master/notebooks/serialization.ipynb), namely `A.to_dict()` and `A.from_dict()`.  For example:
```python3
augmentation = A.Compose([
	A.RandomCrop(768, 768),
	A.OneOf([
		A.RGBShift(),
		A.HueSaturationValue()
	])
])

ChipClassificationConfig(
	...,
	augmentation=A.to_dict(augmentation)
)
```

### PR DESCRIPTION
**Pull request by AdeelH**
## Overview

Adds support for arbitrary albumentations transforms.

### Implementation
- Adds an `aug_transform` option to `DataConfig` that accepts a serialized albumentations transform
https://github.com/azavea/raster-vision/blob/ea53c7ca15dc053f567ab96379a08aeeb1d3a3e3/rastervision_pytorch_learner/rastervision/pytorch_learner/learner_config.py#L279-L284
- **What about augmentations like `A.Normalize()` that need to be applied to the val and test sets too?**
  - Adds a `base_transform` option to `DataConfig` that also accepts a serialized albumentations transform, and is applied to all datasets. Now `A.Normalize()` can be passed in here instead of `aug_transform`.
  https://github.com/azavea/raster-vision/blob/ea53c7ca15dc053f567ab96379a08aeeb1d3a3e3/rastervision_pytorch_learner/rastervision/pytorch_learner/learner_config.py#L272-L278
- **Doesn't `A.Normalize()` mess up the plots because of values no longer being in [0, 1]?**
  - Yes, it does. To undo the normalization (or any other such `base_transform`) before plotting, another transform needs to be provided. Specifically, the following undoes the normalization: 
    ```python3
    plot_transform = A.Normalize(mean=(-mu / std).tolist(), std=(1 / std).tolist(), max_pixel_value=1.)
    ```
  - This is done through adding a `plot_options` option to `DataConfig` that accepts the new `PlotOptions` config.
  https://github.com/azavea/raster-vision/blob/ea53c7ca15dc053f567ab96379a08aeeb1d3a3e3/rastervision_pytorch_learner/rastervision/pytorch_learner/learner_config.py#L285-L286
  https://github.com/azavea/raster-vision/blob/ea53c7ca15dc053f567ab96379a08aeeb1d3a3e3/rastervision_pytorch_learner/rastervision/pytorch_learner/learner_config.py#L214-L224
  - **Note:** This more general solution is expected to be useful in the future for customizing plots further. In particular, the `channel_display_groups` option from #972 can be moved here.
- **What about multiband images?**
  - This works out of the box for multiband images. See `isprs_potsdam.py`.
  - Except for transforms that only work on RGB images such as `A.RGBShift()`, `A.CLAHE()`, `A.HueSaturationValue()`, etc.

### Example usage

```python3
aug_transform = A.Compose([
	A.Flip(),
	A.RandomRotate90(),
	A.CoarseDropout()
	# etc.
])
base_transform = A.Normalize(mean=mu.tolist(), std=std.tolist())
plot_transform = A.Normalize(mean=(-mu / std).tolist(), std=(1 / std).tolist(), max_pixel_value=1.)

backend = PyTorchChipClassificationConfig(
	model=model,
	solver=solver,
	log_tensorboard=True,
	run_tensorboard=False,
	base_transform=A.to_dict(base_transform),
	aug_transform=A.to_dict(aug_transform),
	plot_options=PlotOptions(transform=A.to_dict(plot_transform)))
```
- If `DataConfig.aug_transform` is given, `DataConfig.augmentors` is ignored.
- Examples added to `spacenet_rio.py` and `isprs_potsdam.py`. Enable by passing `-a augment True`.

### Example augmentation result
train

## Checklist

- [x] Updated `docs/changelog.rst`
- [x] Added `needs-backport` label if PR is bug fix that applies to previous minor release
- [x] Ran scripts/format_code and committed any changes
- [x] Documentation updated if needed
- [x] PR has a name that won't get you publicly shamed for vagueness

## Testing Instructions

* How to test this PR
    - Try different Albumentations transforms

Closes #995

### CODE REVIEW

File: rastervision_pytorch_learner/rastervision/pytorch_learner/learner_config.py [hunk 1]
```diff
@@ -210,6 +211,19 @@ def validate_config(self):
                 'class_loss_weights is not supported with external_loss_def.')
 
 
+@register_config('plot_options')
+class PlotOptions(Config):
+    """Config related to plotting."""
+    transform: Optional[dict] = Field(
+        None,
+        description='An Albumentations transform serialized as a dict that '
```
Comment (lewfish) [inline rastervision_pytorch_learner/rastervision/pytorch_learner/learner_config.py:219]: I would just be a little more verbose and add that the main purpose of this is to undo any augmentations that you don't want to be included in the plot, eg. normalization.

**Review by lewfish — CHANGES_REQUESTED** [review]
Looks good aside from minor comment.

**Review by lewfish — APPROVED** [review]
